packages/tx/src/eip4844Transaction.ts

```diff
@@ -0,0 +1,479 @@
+    this._validateCannotExceedMaxInteger({
+      maxFeePerGas: this.maxFeePerGas,
+      maxPriorityFeePerGas: this.maxPriorityFeePerGas,
+    })
+
+    BaseTransaction._validateNotArray(txData)
+
+    if (this.gasLimit * this.maxFeePerGas > MAX_INTEGER) {
+      const msg = this._errorMsg('gasLimit * maxFeePerGas cannot exceed MAX_INTEGER (2^256-1)')
+      throw new Error(msg)
+    }
+
+    if (this.maxFeePerGas < this.maxPriorityFeePerGas) {
+      const msg = this._errorMsg(
+        'maxFeePerGas cannot be less than maxPriorityFeePerGas (The total must be the larger of the two)'
+      )
+      throw new Error(msg)
+    }
+
+    this.maxFeePerDataGas = txData.maxFeePerDataGas ?? BigInt(0)
+
+    this.versionedHashes = txData.versionedHashes ?? []
+    this._validateYParity()
+    this._validateHighS()
+
+    if (this.common.isActivatedEIP(3860)) {
+      checkMaxInitCodeSize(this.common, this.data.length)
+    }
+
+    for (const hash of this.versionedHashes) {
+      if (hash.length !== 32) {
+        const msg = this._errorMsg('versioned hash is invalid length')
+        throw new Error(msg)
+      }
+      if (hash[0] !== BLOB_COMMITMENT_VERSION_KZG) {
+        const msg = this._errorMsg('versioned hash does not start with KZG commitment version')
+        throw new Error(msg)
+      }
+    }
+    if (this.versionedHashes.length > LIMIT_BLOBS_PER_TX) {
+      const msg = this._errorMsg(`tx can contain at most ${LIMIT_BLOBS_PER_TX} blobs`)
+      throw new Error(msg)
+    }
+
+    this.blobs = txData.blobs
+    this.kzgCommitments = txData.kzgCommitments
+
+    const freeze = opts?.freeze ?? true
+    if (freeze) {
+      Object.freeze(this)
+    }
+  }
+
+  public static fromTxData(txData: BlobEIP4844TxData, opts?: TxOptions) {
+    return new BlobEIP4844Transaction(txData, opts)
+  }
+
+  /**
+   * Creates the minimal representation of a blob transaction from the network wrapper version.
+   * The minimal representation is used when adding transactions to an execution payload/block
+   * @param txData a {@link BlobEIP4844Transaction} containing optional blobs/kzg commitments
+   * @param opts - dictionary of {@link TxOptions}
+   * @returns the "minimal" representation of a BlobEIP4844Transaction minus blobs and kzg commitments
+   */
+  public static minimalFromNetworkWrapper(txData: BlobEIP4844Transaction, opts?: TxOptions) {
+    const tx = BlobEIP4844Transaction.fromTxData(
+      {
+        ...txData,
+        ...{ blobs: undefined, kzgCommitments: undefined },
+      },
+      opts
+    )
+    return tx
+  }
+
+  /**
+   * Creates a transaction from the network encoding of a blob transaction (with blobs/commitments/proof)
+   * @param serialized a buffer representing a serialized BlobTransactionNetworkWrapper
+   * @param opts any TxOptions defined
+   * @returns a BlobEIP4844Transaction
+   * @throws if no KZG library is loaded -- using the {@link initKzg} helper method
+   */
+  public static fromSerializedBlobTxNetworkWrapper(
```

Comment:
I guess for me a name like `fromSerializedNetworkEncodedBlobTx()` would work better (also for the other `NetworkWrapper` names in the class (or is `NetworkWrapper` somewhat of an official term in the specs?).

Just dropping here, take or ignore.

Reply:
Side note: then maybe we can/should also drop the `Blob` part since somewhat redundant when called with the class name, also generally just clear from the context.

Reply:
(also let me know if `NetworkEncodedTx` is plain wrong for some reasons)

Reply:
The term `network wrapper` is taken directly from the EIP.   I'm open to changing it though I think it does make sense.  Here's where it comes from.
The flow of blob transactions starts when one enters the mempool and it needs to enter in the `network wrapper` form, so the basic transaction plus the blobs and the kzg commitments.  The EL layer node validates the contents of the network wrapper to verify that the blobs match the kzg commitments and then the versioned hashes (which is just a kzg commitments that's been modified so the first byte is always a `01` (i.e. a version 1 blob transaction).  Once the transaction is included in a block, the blobs and kzg commitments are stripped from the transaction (i.e. removing the network wrapper) and then passed to the CL client via the `getBlobsBundleV1` engine API while the interior transaction (what the EIP refers as the "minimal representation") is included in the block.  Once this transaction has been included in a block, the CL layer is responsible for maintaining the blobs/kzg commitments and the EL client discards the blobs and kzg commitments and just references the transaction by its "minimal representation."   Does this make sense? This why I have separate serialization/deserialization methods for the transaction and then the network wrapper.  I think this addresses several of your questions but will review each of them for any other pertinent details).

Reply:
Thanks for the detailed explanation! 🙏 Ok, then let's stick with the EIP names for now.

I wonder if this naming is optimal in the EIP in the first place, I am still not getting completely warm with it since I think it mixes several things up and naming shouldn't be based on the how things are processed but what things are. So only because the blob-including version is mainly for being transported over the wire (in the current spec) shouldn't lead to use *this* for the name of the TX itself, I would rather unbundle this and call this rather extended/full/maximal version or whatever.

(Maximal vs minimal version sounds more consistent to me than "Network Wrapper" vs. Minimal Verison 😜).

Might take this over as a light discussion point to the interop workshop. 🙂